openedx_events/avro_attrs_bridge_extensions.py

```diff
@@ -0,0 +1,63 @@
+"""
+Classes to serialize and deserialize custom types used by openedx events
+"""
+from abc import ABC, abstractmethod
+from datetime import datetime
+
+from opaque_keys.edx.keys import CourseKey
+
+from openedx_events.avro_attrs_bridge import AVRO_TYPE_FOR
+
+
+class AvroAttrsBridgeExtention(ABC):
+    """
+    Used by openedx_events.avro_attrs_bridge.AvroAttrsBridge class to serialize/deserialize custom types
+    """
+    cls: type
+
+    def type(self):
+        return type(self.cls)
+
+    @abstractmethod
+    def serialize(self, obj) -> bytes:
+        ...
+
+    @abstractmethod
+    def deserialize(self, data: bytes) -> object:
+        ...
+
+    @abstractmethod
+    def record_fields(self):
+        ...
```

Comment:
For reviewer: I choose to write the code for extensions into classes. I could have chooses Named Tuples or even a dict. I am open to alternatives.